src/engine/strat_engine/backstore/devices.rs

```diff
@@ -237,9 +239,32 @@ pub fn initialize_devices(
         dev_info: &DeviceInfo,
         pool_uuid: PoolUuid,
         mda_data_size: MDADataSize,
+        key_description: Option<&str>,
     ) -> StratisResult<StratBlockDev> {
-        let mut f = OpenOptions::new().write(true).open(&dev_info.devnode)?;
+        let dev_uuid = Uuid::new_v4();
+        let (path, deviceno) = match key_description {
+            Some(desc) => {
+                let path = initialize_encrypted_stratis_device(
+                    &dev_info.devnode,
+                    pool_uuid,
+                    dev_uuid,
+                    desc,
+                )?;
+                let dm = DM::new()?;
```

Comment:
We have a perpetual devicemapper context defined in ```src/engine/strat_engine/dm.rs``` which you should use here.

Reply:
Resolved.